packages/react-relay/modern/ReactRelayRefetchContainer.js

```diff
@@ -225,6 +234,10 @@ function createContainerWithFragments<TBase: ReactClass<*>>(
         }
         // TODO t15106389: add helper utility for fetching more data
         this._pendingRefetch = null;
+        this._relayContext = {
+          environment: this.context.relay.environment,
+          variables: fragmentVariables
```

Comment:
The issue wasn't related to what's in `relayContext`. Somehow childContextTypes were missing for certain compositions of containers.